desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -107,19 +107,24 @@ def handle_on_link_shared(channel_id, message_ts, links):
 
     try:
       if path == '/hue/editor' and id_type == 'editor':
-        doc = Document2.objects.get(id=qid)
+        doc = Document2.objects.get(id=qid) if qid.isdigit() else Document2.objects.get(uuid=qid)
         doc_type = 'Query'
       elif path == '/hue/gist' and id_type == 'uuid' and ENABLE_GIST_PREVIEW.get():
         doc = _get_gist_document(uuid=qid)
         doc_type = 'Gist'
       else:
         raise PopupException(_("Cannot unfurl link"))
+      
+      user = User.objects.get(username=doc.owner.username)
+      doc.can_read_or_exception(user)
```

Comment:
Cleaner to move to bit below to keep chunks smaller and not all in one, e.g.

get query
get user
check user perm (critical)
fetch results